lib/spack/llnl/util/filesystem.py

```diff
@@ -242,6 +248,100 @@ def compressed_mime_type(filename):
     return type, subtype
 
 
+@contextmanager
+def stream_in_place(stream):
+    """Ensures that stream is returned to starting position
+    once context manager exits"""
+    current_stream_pos = stream.tell()
+    yield stream
+    stream.seek(current_stream_pos)
+
+
+def compose_bin(bin):
```

Comment:
What is `bin`? A typehint would be useful. I think a name like `bytes_to_int` would work. I think there may be a library method that might do this for you, like `int.from_bytes`.

Reply:
I think that works... Not sure why I reinvented the wheel here...